File: packages/jsii-pacmak/test/expected.jsii-calc/java/src/main/java/software/amazon/jsii/tests/calculator/submodule/child/OuterClass.java
package software.amazon.jsii.tests.calculator.submodule.child;

/**
 * Checks that classes can self-reference during initialization.
 * <p>
 * @see : https://github.com/aws/jsii/pull/1706
 */
@javax.annotation.Generated(value = "jsii-pacmak")
@software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Stable)
@software.amazon.jsii.Jsii(module = software.amazon.jsii.tests.calculator.$Module.class, fqn = "jsii-calc.submodule.child.OuterClass")
public class OuterClass extends software.amazon.jsii.JsiiObject {

    protected OuterClass(final software.amazon.jsii.JsiiObjectRef objRef) {
        super(objRef);
    }

    protected OuterClass(final software.amazon.jsii.JsiiObject.InitializationMode initializationMode) {
        super(initializationMode);
    }

    /**
     */
    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Stable)
    public OuterClass() {
        super(software.amazon.jsii.JsiiObject.InitializationMode.JSII);
        software.amazon.jsii.JsiiEngine.getInstance().createNewObject(this);
    }

    /**
     */
    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Stable)
    public @org.jetbrains.annotations.NotNull software.amazon.jsii.tests.calculator.submodule.child.InnerClass getInnerClass() {
        return this.jsiiGet("innerClass", software.amazon.jsii.tests.calculator.submodule.child.InnerClass.class);
    }
}

File: packages/jsii-pacmak/test/expected.jsii-calc/kotlin/src/main/kotlin/software/amazon/jsii/tests/kotlin/calculator/submodule/child/OuterClass.kt
package software.amazon.jsii.tests.kotlin.calculator.submodule.child

/**
 * Checks that classes can self-reference during initialization.
 * 
 * @see : https://github.com/aws/jsii/pull/1706
 */
@javax.annotation.Generated("jsii-pacmak")
@software.amazon.jsii.Jsii(module = software.amazon.jsii.tests.kotlin.calculator.`$Module`::class, fqn = "jsii-calc.submodule.child.OuterClass")
@software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
open class OuterClass : software.amazon.jsii.JsiiObject {
    constructor(objRef: software.amazon.jsii.JsiiObjectRef) : super(objRef)
    constructor(mode: software.amazon.jsii.JsiiObject.InitializationMode) : super(mode)

    constructor() : super(software.amazon.jsii.JsiiObject.InitializationMode.JSII) {
        software.amazon.jsii.JsiiEngine.getInstance().createNewObject(this)
    }

    @get:software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    open val innerClass: software.amazon.jsii.tests.kotlin.calculator.submodule.child.InnerClass
        get() {
            return this.jsiiGet("innerClass", software.amazon.jsii.tests.kotlin.calculator.submodule.child.InnerClass::class.java) ?: error("'innerClass' should be present")
        }

}
